### PR TITLE
feat: add delete button to edit list page

### PR DESCRIPTION
**Pull request by sourabpramanik**
## Description
This PR has the following:
- a new organism **delete list modal** which can be used for deleting a list.
- The existing delete list modal in the [lists page](https://beta.app.opensauced.pizza/hub/lists) is replaced by **delete list modal organism**  (works like before just some refactoring)
- Added a delete list button at the bottom of the Edit list page which also uses the same **delete list modal organism** 

## What type of PR is this? (check all applicable)

- [X] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [X] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
Feat #2424 

## Mobile & Desktop Screenshots/Recordings
feat

## Steps to QA
1. Go to [Lists page](https://beta.app.opensauced.pizza/hub/lists)
2. Click on any list you want to delete
3. Go to edit list page
4. Scroll to the bottom of the page and you will find the delete list button.

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [X] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?

NO

## [optional] What gif best describes this PR or how it makes you feel?

fun

### CODE REVIEW

**Comment by OgDev-01** [comment]
Thanks for tackling this @sourabpramanik 👏 👏 ... Would you mind adding screen shot/recording of the before and after state of this change?

File: pages/lists/[listId]/edit.tsx [hunk 1]
```diff
@@ -207,133 +216,203 @@ export default function EditListPage({ list, initialContributors }: EditListPage
     );
   };
 
+  // Delete List
```
Comment (nickytonline) [inline pages/lists/[listId]/edit.tsx:219]: refactor (blocking): remove unnecessary comment.

```suggestion
```

File: pages/lists/[listId]/edit.tsx [hunk 2]
```diff
@@ -207,133 +216,203 @@ export default function EditListPage({ list, initialContributors }: EditListPage
...
+              <Text className="my-4">Once you delete a list, you&#39;re past the point of no return.</Text>
+
+              <div>
+                <Button
+                  onClick={() => handleOnDelete()}
+                  variant="default"
+                  className="border bg-light-red-6 border-light-red-8 hover:bg-light-red-7 text-light-red-10"
+                >
+                  Delete list
+                </Button>
```
Comment (nickytonline) [inline pages/lists/[listId]/edit.tsx:401]: refactor (blocking): use the button's `destructive` variant.

```suggestion
                <Button onClick={() => handleOnDelete()} variant="destructive">
                  Delete list
                </Button>
```

Note: Not for this pull request, but the same will need to be done for the Delete Page button in the edit insights page. I'll create an issue for this.

**Review by nickytonline — CHANGES_REQUESTED** [review]
Thanks for the PR @sourabpramanik! A couple change requests.

File: pages/lists/[listId]/edit.tsx (hunk 2 above)
Comment (nickytonline) [reply pages/lists/[listId]/edit.tsx:401]: #2434 has been created for the Delete page button on the edit insights page.

**Comment by sourabpramanik** [comment]
> Thanks for tackling this @sourabpramanik 👏 👏 ... Would you mind adding screen shot/recording of the before and after state of this change?

@OgDev-01 actually before this PR the delete feature was not there in the Edit list page, but I have added the gif for the feature I have created. Will that be okay?

**Comment by sourabpramanik** [comment]
@nickytonline is there any other changes to be done? Please let me know. Thank you for your help.

**Comment by sourabpramanik** [comment]
There are some linting errors. Will resolve them and commit.

**Comment by OgDev-01** [comment]
Testing on a large screen and this is how the button looks
image
We don't want the button spanning the full width of container

**Comment by sourabpramanik** [comment]
Understood will fix this in the next commit.

**Comment by sourabpramanik** [comment]
@OgDev-01 thanks for pointing out, the changes are done

File: pages/lists/[listId]/edit.tsx [hunk 3]
```diff
@@ -390,9 +390,11 @@ export default function EditListPage({ list, initialContributors }: EditListPage
...
               <Text className="my-4">Once you delete a list, you&#39;re past the point of no return.</Text>
 
-              <Button onClick={() => handleOnDelete()} variant="destructive">
-                Delete list
-              </Button>
+              <div>
+                <Button onClick={() => handleOnDelete()} variant="destructive">
+                  Delete list
+                </Button>
+              </div>
```
Comment (OgDev-01) [inline pages/lists/[listId]/edit.tsx:397]: We don't need an extra div here... adding a class of `w-max` to the button should do the trick

```suggestion
      <Button className="w-max" onClick={() => handleOnDelete()} variant="destructive">
                Delete list
       </Button>
```

**Review by OgDev-01 — CHANGES_REQUESTED** [review]

**Comment by OgDev-01** [comment]
You have some lint errors to fix locally @sourabpramanik

File: components/organisms/ListPage/DeleteListPageModal.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+          <Title level={3}>Delete List</Title>
+        </DialogTitle>
+
+        <Text>
+          Are you sure you want to delete <span className="font-bold text-light-slate-12">{listName}</span>? If you have
+          data on this list that your team is using, they will lose access.
+        </Text>
+        <Text>
+          <span className="font-bold text-light-slate-12">This action cannot be undone</span>
+        </Text>
```
Comment (nickytonline) [inline components/organisms/ListPage/DeleteListPageModal.tsx:57]:
```suggestion
        <Text className="font-bold text-light-slate-12">This action cannot be undone</Text>
```

**Review by nickytonline — CHANGES_REQUESTED** [review]
One last small change and this is good to go @sourabpramanik.

File: components/organisms/ListPage/DeleteListPageModal.tsx (hunk 4 above)
Comment (sourabpramanik) [reply components/organisms/ListPage/DeleteListPageModal.tsx:57]: Sure, will be resolved in next commit

**Review by OgDev-01 — APPROVED** [review]
This LGTM 🍕🍕

File: pages/lists/[listId]/edit.tsx [hunk 5]
```diff
@@ -207,133 +216,196 @@ export default function EditListPage({ list, initialContributors }: EditListPage
...
+
+    try {
+      const res = await fetch(`${process.env.NEXT_PUBLIC_API_URL}/lists/${list.id}`, {
+        method: "DELETE",
+        headers: {
+          Authorization: `Bearer ${sessionToken}`,
+        },
+      });
+
+      if (res.ok) {
```
Comment (nickytonline) [inline pages/lists/[listId]/edit.tsx:243]: refactor (blocking): use `fetchApiData`.


```suggestion
      const { error } = await fetchApiData({
        path: "/lists/${list.id}",
        method: "DELETE",
        bearerToken: sessionToken,
        pathValidator: () => true,
      });

      if (!error)
```

File: pages/lists/[listId]/edit.tsx [hunk 6]
```diff
@@ -18,6 +22,9 @@ import Pagination from "components/molecules/Pagination/pagination";
 import { Avatar } from "components/atoms/Avatar/avatar-hover-card";
 import useFetchAllListContributors from "lib/hooks/useFetchAllListContributors";
 
+// lazy import DeleteListPageModal component to optimize bundle size they don't load on initial render
```
Comment (nickytonline) [inline pages/lists/[listId]/edit.tsx:25]: nit: the comment isn't necessary

**Review by nickytonline — CHANGES_REQUESTED** [review]
This is working well @sourabpramanik. A couple change requests.

**Review by nickytonline — DISMISSED** [review]
Thanks for the addition @sourabpramanik!

:shipit:

**Review by nickytonline — CHANGES_REQUESTED** [review]
I prematurely approved and didn't realize there were some build errors.

File: pages/lists/[listId]/edit.tsx [hunk 7]
```diff
@@ -207,133 +215,194 @@ export default function EditListPage({ list, initialContributors }: EditListPage
...
+
+  const handleDeleteModalClose = () => {
+    setIsDeleteModalOpen(false);
+  };
+
+  const handleOnDeleteConfirm = async () => {
+    setSubmitted(true);
+    setDeleteLoading(true);
+
+    const { error } = await fetchApiData({
```
Comment (nickytonline) [inline pages/lists/[listId]/edit.tsx:234]: There's a couple build errors at the moment.

```
./pages/lists/[listId]/edit.tsx:239:9
Type error: Type 'string | null | undefined' is not assignable to type 'string'.
  Type 'undefined' is not assignable to type 'string'.

  237 |         path: "/lists/${list.id}",
  238 |         method: "DELETE",
> 239 |         bearerToken: sessionToken,
      |         ^
  240 |         pathValidator: () => true,
  241 |       });
  242 | 
```
and

```
./pages/lists/[listId]/edit.tsx
248:4  Error: Parsing error: 'catch' or 'finally' expected.
```

Comment (sourabpramanik) [reply pages/lists/[listId]/edit.tsx:234]: I resolved those in my last commit can you please check once?

**Review by nickytonline — COMMENTED** [review]
When I go to delete the list, I get this error when it's doing the delete.

CleanShot 2024-01-15 at 11 22 07

**Comment by sourabpramanik** [comment]
@nickytonline is there any path issue? I suspect the new changes are causing some issues. I will check

**Comment by sourabpramanik** [comment]
> CleanShot 2024-01-15 at 11 22 07

Aaahh!! such a dumb mistake by me. 😂😂😂

**Review by nickytonline — APPROVED** [review]
Works great now! Thanks @sourabpramanik!

:shipit:

**Comment by OgDev-01** [comment]
Great job @sourabpramanik 👏 👍 👌

**Comment by sourabpramanik** [comment]
Thanks @nickytonline and @OgDev-01

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**